core/src/commonMain/kotlin/com/ustadmobile/core/domain/validateusername/ValidateUsernameUseCase.kt

```diff
@@ -0,0 +1,41 @@
+package com.ustadmobile.core.domain.validateusername
+
+/**
+ * Validates whether a username meets all required criteria:
+ * - Must not be too short or too long
+ * - Must not start with a number
+ * - Must only contain valid characters (letters, numbers, dots, underscores)
+ */
+class ValidateUsernameUseCase {
+
+    enum class ValidationResult {
+        VALID,
+        INVALID_TOO_SHORT,
+        INVALID_TOO_LONG,
+        INVALID_STARTS_WITH_NUMBER,
+        INVALID_OTHER
+    }
+
+    operator fun invoke(username: String): ValidationResult {
+        return when {
+            username.length < MIN_LENGTH -> ValidationResult.INVALID_TOO_SHORT
+            username.length > MAX_LENGTH -> ValidationResult.INVALID_TOO_LONG
+            username.firstOrNull()?.isDigit() == true -> ValidationResult.INVALID_STARTS_WITH_NUMBER
+            !username.all { it.isValidUsernameChar() } -> ValidationResult.INVALID_OTHER
+            else -> ValidationResult.VALID
+        }
+    }
+
+    private fun Char.isValidUsernameChar(): Boolean = when {
```

Comment:
shouldBlockKeyEvent =  NOT (isValidUsernameChar)

So why are there two functions with the same logic, **_AGAIN_** (same problem as per [comment on previous pull request](https://github.com/UstadMobile/UstadMobile/pull/986))?

Have one function in ValidateUsernameUseCase as follows:

```
companion object {
    fun isValidUsernameChar(character: Char) {
         return when {
             character.isLetter() -> true
             character.isDigit() -> true
             character in ALLOWED_SPECIAL -> true
             else -> false 
         }
    }

}
```

Remove ```FilterUsernameUseCase.shouldBlockKeyEvent``` - replace this with using ```!ValidateUsernameUseCase.isValidUsernameChar ```